File: 백준/Silver/10773. 제로/제로.cc
#include <iostream>
#include <stack>

using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	int n;
	cin >> n;

	stack<int> st;
	int sum = 0;
	while (n--) {
		int temp;
		cin >> temp;
		if (temp == 0) {
			sum -= st.top();
			st.pop();
		}
		else {
			st.push(temp);
			sum += temp;
		}
	}
	cout << sum;
}

File: 백준/Silver/10773. 제로/제로.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        Stack<Integer> stack = new Stack<>();
        
        for (int i = 0; i < N; i++) {
            int a = sc.nextInt();
            if (a != 0){
                stack.push(a);
            } else if (!stack.isEmpty()) {
                stack.pop();
            }
        }
        int sum = 0; // 스택에 남아 있는 숫자의 합 계산
        for (int i = 0; i < stack.size(); i++) {
            sum += stack.get(i);
        }
        System.out.println(sum);
    }
}
